keras/utils/generic_utils.py

```diff
@@ -296,7 +296,7 @@ def update(self, current, values=None, force=False):
                 return
 
             prev_total_width = self.total_width
-            if sys.stdout.isatty():
+            if sys.stdout.isatty() or 'ipykernel' in sys.modules:
```

Comment:
This will search for a string match in a long list. Not something we want to do at every update. Let's make it a boolean progbar attribute set at init.

Also: is this the best way to check whether we should print `\b`s or not?

Reply:
There isn't any real good way to do it. Jupyter is a special case it seems where they are using 0mq for stdout to the browser. It isn't a tty just stdout. In python it seems that it's either tty or everything else.
I don't know how important it is to pipe the output to a file, you can always implement a callback that writes information about the current run to a file.

If it were up to me I would probably revert #8171 and close this PR.